test/unit/jwt-token-manager.test.js

```diff
@@ -141,14 +147,86 @@ describe('JWT Token Manager', () => {
       expect(requestTokenSpy).toHaveBeenCalled();
       expect(saveTokenInfoSpy).toHaveBeenCalled();
       expect(decode).toHaveBeenCalled();
+      expect(token).toBe(ACCESS_TOKEN);
+      expect(instance.accessToken).toBe(ACCESS_TOKEN);
+      expect(tokenNeedsRefreshSpy).not.toHaveBeenCalled();
+
+      saveTokenInfoSpy.mockRestore();
+      decode.mockRestore();
+      requestTokenSpy.mockRestore();
+      tokenNeedsRefreshSpy.mockRestore();
+    });
+
+    it('should refresh token if token is stored but needs refresh (close to expired)', async () => {
```

Comment:
This now checks the "needs refresh but hasn't expired" path (the success version)